### PR TITLE
Feature/eui 1882 session timeouts

### PR DESCRIPTION
**Pull request by ficklephil**
Adding the Session Timeout Modal.
Session Timeouts calculated in the Node layer.
Use of different Session Timeouts for different User on Manage Cases, set via configuration.

For a selection of timeout times please see the following Jira ticket.
https://tools.hmcts.net/jira/browse/EUI-1882

### CODE REVIEW

File: src/app/containers/app/app.component.ts [hunk 1]
```diff
@@ -1,16 +1,191 @@
...
+   *  "pattern": "-solicitor",
+   *  "totalIdleTime": 50
+   *  }
+   * }
+   */
+  public userDetailsHandler(userDetails) {
+
+    if (propsExist(userDetails,['sessionTimeout']) && userDetails.sessionTimeout.totalIdleTime > 0) {
+
+      console.log(userDetails.sessionTimeout);
```
Comment (ficklephil) [inline src/app/containers/app/app.component.ts:67]: Temporary, making it easier for QA to test.

File: api/user/index.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+interface SessionRequest extends express.Request {
+  session: {
+    user: {
+      roles: ['']
+    }
+  },
+}
+
+// TODO: Need to test error condition here.
+export async function getUserDetails(req: SessionRequest, res: express.Response, next: express.NextFunction) {
```
Comment (paddybasi) [inline api/user/index.ts:15]: could we not do this please? I never liked the enhancedRequest etc before either. Instead try doing a declaration merging, but only if the @types aren't available

File: src/app/services/auth/auth.service.ts [hunk 3]
```diff
@@ -81,8 +81,20 @@ export class AuthService {
...
     window.location.href = '/api/logout';
   }
+
+  logOutAndRedirect() {
+
+    console.log('logOutAndRedirect');
+    const { hostname, port } = window.location;
+    const portNumber = port ? `:${port}` : '';
+
+    const baseUrl = `http://${hostname}${portNumber}`;
```
Comment (paddybasi) [inline src/app/services/auth/auth.service.ts:95]: please don't do this, let the backend do it? also protocol will be wrong etc, not always http

File: api/user/index.ts (hunk 2 above)
Comment (ficklephil) [reply api/user/index.ts:15]: Fixed after call with Paddy.

File: src/app/services/auth/auth.service.ts (hunk 3 above)
Comment (ficklephil) [reply src/app/services/auth/auth.service.ts:95]: Awaiting the Open Id logout route, then we will refactor this as discussed with Paddy.

**Pull request merged**